### ISSUE TITLE
Receiver interface undefined behaviors

### ISSUE THREAD

**Issue by Lasering** [ocpi/ocpi#406]
Using the Locations module as an example:

#### `GET <base-uri> / countryCode / partyId / locationId / evseUid`

If locationId refers to a location that does not exist should a HTTP 404 be returned? (I assume yes, just wanted a confirmation). This is even more pertinent if evseUid refers to an existing evseUid.

#### `PUT <base-uri> / countryCode / partyId / locationId`

What is correct implementation in the following cases:

1. countryCode and/or partyId in the URL are different from the Location parsed from the request body? Does this mean the CPO has transferred ownership of the Location to another CPO?

    - If yes, should the location that existed in the previous countryCode/partyId/locationId be deleted? 
    - If not, should a OCPI status code 2001 be returned?

2. locationId in the URL is different from the parsed Location id. Given that the location id is described by:

    > Uniquely identifies the location within the CPOs platform (and suboperator platforms). This field can never be changed, modified or renamed.

    I guess this should be an error, but which OCPI status code should be returned 2001?

#### `PUT <base-uri> / countryCode / partyId / locationId / evseUid`

If locationId refers to a location that does not exist should a HTTP 404 be returned? (almost the same question as for the GET)

**Comment by reinierl** [comment]
> Using the Locations module as an example:
> #### `GET <base-uri> / countryCode / partyId / locationId / evseUid`
> 
> If locationId refers to a location that does not exist should a HTTP 404 be returned? (I assume yes, just wanted a confirmation). This is even more pertinent if evseUid refers to an existing evseUid.
> #### `PUT <base-uri> / countryCode / partyId / locationId`

Yes, there should be an HTTP 404 as no resource with that URI exists.

> What is correct implementation in the following cases:
> 
>     1. countryCode and/or partyId in the URL are different from the Location parsed from the request body? Does this mean the CPO has transferred ownership of the Location to another CPO?
>        
>        * If yes, should the location that existed in the previous countryCode/partyId/locationId be deleted?
>        * If not, should a OCPI status code 2001 be returned?

The OCPI specification seems to allow you to use different values of country code and party ID in the URL  from the ones you use in the object iself. It doesn't seem to say what it means if a client does that.

One possibility I considered is that it might be useful for objects from roaming hubs, where roaming hubs would perhaps be sending objects to a URL with the hub's party ID and the object issuer's party ID in the body. But OCPI makes the Hub use the object issuer's party ID in the URL as well: https://github.com/ocpi/ocpi/blob/release-2.2.1-bugfixes/mod_hub_client_info.asciidoc#111-another-party-becomes-connected .

@RobertDeLeeuw do you know what the intended use of the country code and party ID fields in the object itself is?

>     2. locationId in the URL is different from the parsed Location id. Given that the location id is described by:
>        > Uniquely identifies the location within the CPOs platform (and suboperator platforms). This field can never be changed, modified or renamed.
>        
>        
>        I guess this should be an error, but which OCPI status code should be returned 2001?

Yes, that's what the standard "advises": https://github.com/ocpi/ocpi/blob/release-2.2.1-bugfixes/transport_and_format.asciidoc#errors

> #### `PUT <base-uri> / countryCode / partyId / locationId / evseUid`
> 
> If locationId refers to a location that does not exist should a HTTP 404 be returned? (almost the same question as for the GET)

Yes. The resource doesn't exist so you can't `PUT` on it.
